### ISSUE TITLE
Malformed .vela.yml Panic

### ISSUE THREAD

**Issue by GregoryDosh** [go-vela/community#300]
## Description
A specifically formed .vela.yml file can cause a panic on the Vela CLI, and I suspect it'll impact other aspects of the Vela ecosystem such as workers or anything that tries to parse the .vela.yml.

Example .vela.yml
```yaml
version: "1"

services:
  - name: "YAML Panic"
    environment:
     - 'This: Panics'
```

## Value
This causes issues on the CLI because one can't write `vela validate pipeline` and I suspect it is causing CI environment stability issues if people are sending malformed YAML files since it's a hard panic that isn't getting caught.

## Useful Information
1. What is the output of `vela --version`?
`vela version v0.8.1`

2. What operating system is being used?
macOS Big Sur 11.3.1 (20E241)

3. Any other important details?
Here is the stack trace of the panic
```sh
vela validate pipeline
```
```sh
panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1 [recovered]
	panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/buildkite/yaml.handleErr(0xc00061f5c0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/yaml.go:249 +0x8d
panic(0x271ae40, 0xc000046570)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/buildkite/yaml.handleErr(0xc00061ef48)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/yaml.go:249 +0x8d
panic(0x271ae40, 0xc000046570)
	/usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/go-vela/types/raw.(*StringSliceMap).UnmarshalYAML(0xc0001c0888, 0xc0000dfbc0, 0x0, 0x8441ff8)
	/go/pkg/mod/github.com/go-vela/types@v0.8.2/raw/map.go:105 +0x3ef
github.com/buildkite/yaml.(*decoder).callUnmarshaler(0xc0007aaf00, 0xc0001c0700, 0x8441ff8, 0xc0001c0888, 0xc0001c0888)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:262 +0xa6
github.com/buildkite/yaml.(*decoder).prepare(0xc0007aaf00, 0xc0001c0700, 0x2634e40, 0xc0001c0888, 0x195, 0xc000803bb8, 0xc0000b1bd0, 0xc000803b80, 0x2a40000)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:305 +0x1d0
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c0700, 0x2634e40, 0xc0001c0888, 0x195, 0xc0001c0888)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:320 +0x6e
github.com/buildkite/yaml.(*decoder).mappingStruct(0xc0007aaf00, 0xc0001c0540, 0x2726be0, 0xc0001c0850, 0x199, 0x25a4548)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:732 +0xa33
github.com/buildkite/yaml.(*decoder).mapping(0xc0007aaf00, 0xc0001c0540, 0x2726be0, 0xc0001c0850, 0x199, 0x2726be0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:573 +0xa67
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c0540, 0x25a4500, 0xc0004074b8, 0x196, 0x196)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:328 +0x133
github.com/buildkite/yaml.(*decoder).sequence(0xc0007aaf00, 0xc0001c0460, 0x24d0880, 0xc00000c7b0, 0x197, 0x24d0880)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:558 +0x252
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c0460, 0x24a5fc0, 0xc00000c7b0, 0x16, 0x18)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:330 +0x10a
github.com/buildkite/yaml.(*decoder).callUnmarshaler.func1(0x24a5fc0, 0xc00000c7b0, 0x0, 0x0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:264 +0x118
github.com/go-vela/types/yaml.(*ServiceSlice).UnmarshalYAML(0xc000207498, 0xc0000dfa60, 0x0, 0x8440f78)
	/go/pkg/mod/github.com/go-vela/types@v0.8.2/yaml/service.go:64 +0x50
github.com/buildkite/yaml.(*decoder).callUnmarshaler(0xc0007aaf00, 0xc0001c0460, 0x8440f78, 0xc000207498, 0xc000207498)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:262 +0xa6
github.com/buildkite/yaml.(*decoder).prepare(0xc0007aaf00, 0xc0001c0460, 0x2576740, 0xc000207498, 0x197, 0xc000803908, 0xc0000b1b70, 0xc0008038c0, 0x2a40000)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:305 +0x1d0
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c0460, 0x2576740, 0xc000207498, 0x197, 0xc000207498)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:320 +0x6e
github.com/buildkite/yaml.(*decoder).mappingStruct(0xc0007aaf00, 0xc0001c02a0, 0x274bb00, 0xc000207440, 0x199, 0x0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:732 +0xa33
github.com/buildkite/yaml.(*decoder).mapping(0xc0007aaf00, 0xc0001c02a0, 0x274bb00, 0xc000207440, 0x199, 0x274bb00)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:573 +0xa67
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c02a0, 0x274bb00, 0xc000207440, 0x199, 0xc0001c02a0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:328 +0x133
github.com/buildkite/yaml.(*decoder).document(0xc0007aaf00, 0xc0001c0230, 0x274bb00, 0xc000207440, 0x199, 0x600)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:340 +0x7c
github.com/buildkite/yaml.(*decoder).unmarshal(0xc0007aaf00, 0xc0001c0230, 0x274bb00, 0xc000207440, 0x199, 0x199)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/decode.go:316 +0x1c5
github.com/buildkite/yaml.unmarshal(0xc00020e600, 0x56, 0x200, 0x24b1000, 0xc000207440, 0x2a6a200, 0x0, 0x0)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/yaml.go:148 +0x352
github.com/buildkite/yaml.Unmarshal(...)
	/go/pkg/mod/github.com/buildkite/yaml@v0.0.0-20181016232759-0caa5f0796e3/yaml.go:81
github.com/go-vela/compiler/compiler/native.ParseBytes(0xc00020e600, 0x56, 0x200, 0x56, 0x200, 0x0)
	/go/pkg/mod/github.com/go-vela/compiler@v0.8.1/compiler/native/parse.go:47 +0x74
github.com/go-vela/compiler/compiler/native.ParseReader(0x2a6bcc0, 0xc000407490, 0x0, 0x0, 0xc000407490)
	/go/pkg/mod/github.com/go-vela/compiler@v0.8.1/compiler/native/parse.go:81 +0xe5
github.com/go-vela/compiler/compiler/native.ParsePath(0xc00003eb40, 0x32, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/go-vela/compiler@v0.8.1/compiler/native/parse.go:70 +0x1a6
github.com/go-vela/compiler/compiler/native.(*client).Parse(0xc0002c10e0, 0x251dda0, 0xc0000b1b40, 0x13, 0xc00061f7b0, 0x1)
	/go/pkg/mod/github.com/go-vela/compiler@v0.8.1/compiler/native/parse.go:32 +0x1c5
github.com/go-vela/cli/action/pipeline.(*Config).ValidateLocal(0xc00061f8a8, 0x2ac2690, 0xc0002c10e0, 0x0, 0x6)
	/github/workspace/action/pipeline/validate.go:99 +0x22b
github.com/go-vela/cli/action.pipelineValidate(0xc0007aae00, 0x7, 0xc)
	/github/workspace/action/pipeline_validate.go:146 +0x31c
github.com/urfave/cli/v2.(*Command).Run(0x3622240, 0xc0007aaa00, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x4dd
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc00039d6c0, 0xc0007aa380, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:434 +0xa49
github.com/urfave/cli/v2.(*Command).startApp(0x3624880, 0xc0007aa380, 0x7ffeefbff505, 0x8)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:278 +0x69b
github.com/urfave/cli/v2.(*Command).Run(0x3624880, 0xc0007aa380, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:94 +0x98d
github.com/urfave/cli/v2.(*App).RunContext(0xc00039d1e0, 0x2aa1570, 0xc0000d0160, 0xc0000b2180, 0x3, 0x3, 0x0, 0x0)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0x810
github.com/urfave/cli/v2.(*App).Run(...)
	/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
	/github/workspace/cmd/vela-cli/main.go:128 +0xe05
```

**Comment by colindean** [comment]
I've lost track of the number of times I've bumped my head on this when rolling out a new config. It's usually because I've specified `environment` as

```yaml
environment:
  - key: value
```

or 

```yaml
environment:
  - key=value
```

like how some other CI/CD YAML defines variables. It'd be nice if Vela did one of these:

1. Accept all of these formats:
    * YAML property
    * `:`-separated key-value pairs in an array
    * `=`-separated key-value pairs in an array
2. Warn with a suggestion of how to correct it when it encounters one of the invalid formats.

I think № 2 is more appropriate and less confusing in the long run. 

I thought I'd filed this previously but I guess not! Thanks for doing so, @GregoryDosh!
